lib/search.js

```diff
@@ -12,8 +12,21 @@ module.exports = function(appConfig, date_field) {
      date
      type
      */
+
+    function validate(res, data, type, errField) {
+        if (typeof data !== type) {
+            res.status(500).json({error: "the " + errField + " parameter must be a " + type + ", was given: " + data});
+            return false;
+        }
+        return true;
+    }
+
     function performSearch(context, req, res, config) {
-        if (req.query.size > 100000) {
+        if (req.query.size) {
+            if (!validate(res, req.query.size, "number", "size")) return;
```

Comment:
Don't think this is working correctly.

{"error":"the size parameter must be a number, was given: 1"}